src/audio_player.rs

```diff
@@ -0,0 +1,90 @@
+use crate::sync_wrapped;
+use bevy::prelude::{AudioSinkPlayback, Component, Vec3};
+
+#[derive(Component)]
+pub struct AudioListener {
+    pub previous_position: Vec3,
+}
+
+impl Default for AudioListener {
+    fn default() -> Self {
+        AudioListener {
+            previous_position: Vec3::default(),
+        }
+    }
+}
+
+#[derive(Component)]
+pub struct AudioSource {
```

Comment:
Just thinking out loud: Can AudioSource and AudioSourcePlayer be merged into one struct? I don't see much reason in separating them right now.

Reply:
Let's say it's combined into AudioSource, then the moment the user spawns this the EventInstance will need to be initialized, which means you need a reference to the Studio. I tried but I'm not sure how to do it without making the example more complicated